### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Vo7kov**
https://vo7kov.github.io/react_movies-list-filter-js/

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,33 +1,52 @@
...
-            />
+function getPrepairedMovies(query) {
+  const prepairedQuery = query.toLowerCase().trim();
+
+  return moviesFromServer.filter((movie) => {
+    const { title, description } = movie;
+
+    return title.trim().toLowerCase().includes(prepairedQuery)
+      || description.trim().toLowerCase().includes(prepairedQuery);
+  });
```
Comment (IvanFesenko) [inline src/App.jsx:14]: No need to filter in case query is empty

File: src/App.jsx [hunk 2]
```diff
@@ -1,33 +1,52 @@
...
+                }}
+              />
+            </div>
           </div>
         </div>
-      </div>
 
-      <MoviesList movies={moviesFromServer} />
-    </div>
+        <MoviesList movies={getPrepairedMovies(query)} />
```
Comment (IvanFesenko) [inline src/App.jsx:44]: Create separate const `visibleMovies`

File: src/App.jsx [hunk 3]
```diff
@@ -1,33 +1,52 @@
...
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={(event) => {
+                  setQuery(event.target.value);
+                }}
```
Comment (IvanFesenko) [inline src/App.jsx:38]: Try do not use anonymous functions - create separate handler

**Review by demetrxx — APPROVED** [review]
Well done 👍 Go ahead!

**Review by IvanFesenko — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 4]
```diff
@@ -1,33 +1,58 @@
...
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+function getPrepairedMovies(query) {
+  let visibleMovies = [...moviesFromServer];
+  const prepairedQuery = query.toLowerCase().trim();
+
+  if (query.length === 0) {
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:10]:
```suggestion
  if (!query) {
```
empty string is falsy value

File: src/App.jsx [hunk 5]
```diff
@@ -1,33 +1,58 @@
...
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:36]:
```suggestion
            <label htmlFor="search-query" className="label" htmlFor="search-query">
              Search movie
            </label>
```
it should fix such eslint issue

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Well done

**Pull request closed without merging**